### Museum Analysis for the MET

Source:  https://github.com/metmuseum/openaccess  

In [1]:
from plotnine import *
import pandas as pd
import matplotlib.pyplot as plt

plt.rcParams['svg.fonttype'] = 'none'

/Users/misha/.pyenv/versions/3.10.0/lib/python3.10/site-packages/pandas/compat/__init__.py:124: UserWarning: Could not import the lzma module. Your installed Python is incomplete. Attempting to use lzma compression will result in a RuntimeError.
  warnings.warn(msg)


### Let's start by importing and cleaning the data 

In [2]:
df = pd.read_csv("MetObjects.csv", low_memory=False, encoding = 'utf-8')

In [3]:
df.columns = df.columns.str.replace(' ', '_')

### Because the MET has over 470,000 artworks, we are only going to focus on the paintings so we can get the gender for a larger proportion using the Artsy API

In [4]:
df = df.filter(['Title', 'Artist_Display_Name', 'Artist_Gender',  'Object_Name', 'Department'])

In [5]:
df = df[df.Object_Name == 'Painting']

In [6]:
df

Title             Artist_Display_Name  \
409     Plaque Portrait of Benjamin Franklin  Louis Benjamin Huber Defeuille   
410     Plaque Portrait of Benjamin Franklin         Jacques Thouron|Weyller   
411     Plaque Portrait of Benjamin Franklin                 Jacques Thouron   
412     Plaque Portrait of Benjamin Franklin                 Jacques Thouron   
413     Plaque Portrait of Benjamin Franklin                             NaN   
...                                      ...                             ...   
477154                           Cool Topics                    Julia Rommel   
477310                                   NaN             Unidentified artist   
477483                                   NaN             Unidentified artist   
477484                                   NaN             Unidentified artist   
477485                                   NaN             Unidentified artist   

       Artist_Gender Object_Name                   Department  
409              NaN    Painting            The American Wing  
410                |    Painting            The American Wing  
411              NaN    Painting            The American Wing  
412              NaN    Painting            The American Wing  
413              NaN    Painting            The American Wing  
...              ...         ...                          ...  
477154           NaN    Painting  Modern and Contemporary Art  
477310           NaN    Painting                    Asian Art  
477483           NaN    Painting                    Asian Art  
477484           NaN    Painting                    Asian Art  
477485           NaN    Painting                    Asian Art  

[5932 rows x 5 columns]

In [7]:
df = df[df.Department == 'Modern and Contemporary Art']

In [8]:
df

Title  Artist_Display_Name Artist_Gender Object_Name  \
87625   Berthe David-Weill        Salvador Dalí           NaN    Painting   
322498       Self-Portrait  William Rothenstein           NaN    Painting   
322508       Up the Hudson       George Bellows           NaN    Painting   
322512        October Seas       Paul Dougherty           NaN    Painting   
322513       Morning Light      Eugene Speicher           NaN    Painting   
...                    ...                  ...           ...         ...   
474852     Contour of Loss         Titus Kaphar           NaN    Painting   
475110                Jugs        Meyer Vaisman           NaN    Painting   
475910            Untitled        William Scott           NaN    Painting   
477108          The Bather   Thomas Hart Benton           NaN    Painting   
477154         Cool Topics         Julia Rommel           NaN    Painting   

                         Department  
87625   Modern and Contemporary Art  
322498  Modern and Contemporary Art  
322508  Modern and Contemporary Art  
322512  Modern and Contemporary Art  
322513  Modern and Contemporary Art  
...                             ...  
474852  Modern and Contemporary Art  
475110  Modern and Contemporary Art  
475910  Modern and Contemporary Art  
477108  Modern and Contemporary Art  
477154  Modern and Contemporary Art  

[1872 rows x 5 columns]

### We have to make a new column so we can match the artist names in the Artsy database.

In [9]:
df['artist'] = df['Artist_Display_Name'].str.lower()
df['artist'] = df['artist'].str.replace(" ", "-")
df['artist'] = df['artist'].str.replace("after-", "")
df['artist'] = df['artist'].str.replace(".", "")
df['artist'] = df['artist'].str.replace("é", "e")
df['artist'] = df['artist'].str.replace("è", "e")
df['artist'] = df['artist'].str.replace("è", "e")
df['artist'] = df['artist'].str.replace("&", "and")
df['artist'] = df['artist'].str.replace("í", "i")
df['artist'] = df['artist'].str.replace("á", "a")
df['artist'] = df['artist'].str.replace("attributed-to-", "")
df['artist'] = df['artist'].str.replace("'", "")
df['artist'] = df['artist'].str.replace(",", "")
df['artist'] = df['artist'].str.replace("’", "")

/var/folders/7j/5h5yf6k54nz1ys4trym9bls40000gn/T/ipykernel_71020/3787383773.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
/var/folders/7j/5h5yf6k54nz1ys4trym9bls40000gn/T/ipykernel_71020/3787383773.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
/var/folders/7j/5h5yf6k54nz1ys4trym9bls40000gn/T/ipykernel_71020/3787383773.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the docume

In [10]:
df

Title  Artist_Display_Name Artist_Gender Object_Name  \
87625   Berthe David-Weill        Salvador Dalí           NaN    Painting   
322498       Self-Portrait  William Rothenstein           NaN    Painting   
322508       Up the Hudson       George Bellows           NaN    Painting   
322512        October Seas       Paul Dougherty           NaN    Painting   
322513       Morning Light      Eugene Speicher           NaN    Painting   
...                    ...                  ...           ...         ...   
474852     Contour of Loss         Titus Kaphar           NaN    Painting   
475110                Jugs        Meyer Vaisman           NaN    Painting   
475910            Untitled        William Scott           NaN    Painting   
477108          The Bather   Thomas Hart Benton           NaN    Painting   
477154         Cool Topics         Julia Rommel           NaN    Painting   

                         Department               artist  
87625   Modern and Contemporary Art        salvador-dali  
322498  Modern and Contemporary Art  william-rothenstein  
322508  Modern and Contemporary Art       george-bellows  
322512  Modern and Contemporary Art       paul-dougherty  
322513  Modern and Contemporary Art      eugene-speicher  
...                             ...                  ...  
474852  Modern and Contemporary Art         titus-kaphar  
475110  Modern and Contemporary Art        meyer-vaisman  
475910  Modern and Contemporary Art        william-scott  
477108  Modern and Contemporary Art   thomas-hart-benton  
477154  Modern and Contemporary Art         julia-rommel  

[1872 rows x 6 columns]

In [15]:
import requests
import json

client_id = '0462654e9118c822403d'
client_secret = '5f6c4328fb1499d26260c986f6d2ac12'

# initiate the request to get the token 
r = requests.post("https://api.artsy.net/api/tokens/xapp_token",
                  data={
                      "client_id": client_id,
                      "client_secret": client_secret
                  })

# parse the server response 
j = json.loads(r.text)
j
# extract the token 

{'type': 'xapp_token',
 'token': 'eyJhbGciOiJIUzI1NiJ9.eyJyb2xlcyI6IiIsInN1YmplY3RfYXBwbGljYXRpb24iOiI2MjczZWJjZTg2YTM5NjAwMGM1MjdmYjYiLCJleHAiOjE2NTMzMjUyMTAsImlhdCI6MTY1MjcyMDQxMCwiYXVkIjoiNjI3M2ViY2U4NmEzOTYwMDBjNTI3ZmI2IiwiaXNzIjoiR3Jhdml0eSIsImp0aSI6IjYyODI4MzFhMjQ1ZmY3MDAwZWI1YWNhYSJ9._EJTnnKFXVNZgaMq3ujHCDheBdSloBWcsb4ILY2s-uY',
 'expires_at': '2022-05-23T17:00:10+00:00',
 '_links': {}}

In [16]:
# create a header containing our token 
headers = {"X-Xapp-Token" : 'eyJhbGciOiJIUzI1NiJ9.eyJyb2xlcyI6IiIsInN1YmplY3RfYXBwbGljYXRpb24iOiI2MjczZWJjZTg2YTM5NjAwMGM1MjdmYjYiLCJleHAiOjE2NTMzMjUyMTAsImlhdCI6MTY1MjcyMDQxMCwiYXVkIjoiNjI3M2ViY2U4NmEzOTYwMDBjNTI3ZmI2IiwiaXNzIjoiR3Jhdml0eSIsImp0aSI6IjYyODI4MzFhMjQ1ZmY3MDAwZWI1YWNhYSJ9._EJTnnKFXVNZgaMq3ujHCDheBdSloBWcsb4ILY2s-uY'}
# initiate a request with the header 



r = requests.get("https://api.artsy.net/api/artists/a.g.-rizzoli'", headers=headers)

# parse the server response 
j = json.loads(r.text)
j

JSONDecodeError: Expecting value: line 1 column 1 (char 0)

In [17]:
import time 

case_list = []

errors = []
for name in df['artist'].unique():
        print(name)
        try:
            url = "https://api.artsy.net/api/artists/" + name  #name is the name of the artist we're looking for 
            url = f"{url}"  #puts quotes aruond url 
            print(url)
            r = requests.get(url, headers=headers)
            j = json.loads(r.text)
            
            
            artist = j['slug']
       

            case = {'artist': j['slug'], 'gender': j['gender'], 'nationality': j['nationality'], 'birthday': j['birthday'], 'hometown': j['hometown']}
            case_list.append(case)
            time.sleep(0.1)
        except:
            print("error", name)
            errors.append(name)

salvador-dali
https://api.artsy.net/api/artists/salvador-dali
william-rothenstein
https://api.artsy.net/api/artists/william-rothenstein
george-bellows
https://api.artsy.net/api/artists/george-bellows
paul-dougherty
https://api.artsy.net/api/artists/paul-dougherty
eugene-speicher
https://api.artsy.net/api/artists/eugene-speicher
error eugene-speicher
edward-j-steichen
https://api.artsy.net/api/artists/edward-j-steichen
error edward-j-steichen
jonas-lie
https://api.artsy.net/api/artists/jonas-lie
william-orpen
https://api.artsy.net/api/artists/william-orpen
error william-orpen
rockwell-kent
https://api.artsy.net/api/artists/rockwell-kent
charles-shepard-chapman
https://api.artsy.net/api/artists/charles-shepard-chapman
josef-albers
https://api.artsy.net/api/artists/josef-albers
giorgio-de-chirico
https://api.artsy.net/api/artists/giorgio-de-chirico
maurice-utrillo
https://api.artsy.net/api/artists/maurice-utrillo
error maurice-utrillo
emil-nolde
https://api.artsy.net/api/artists/emil-nold

jim-dine
https://api.artsy.net/api/artists/jim-dine
error jim-dine
françois-rouan
https://api.artsy.net/api/artists/françois-rouan
error françois-rouan
audrey-flack
https://api.artsy.net/api/artists/audrey-flack
jean-isy-de-botton
https://api.artsy.net/api/artists/jean-isy-de-botton
walt-kuhn
https://api.artsy.net/api/artists/walt-kuhn
error walt-kuhn
maurice-golubov
https://api.artsy.net/api/artists/maurice-golubov
malcolm-morley
https://api.artsy.net/api/artists/malcolm-morley
milton-resnick
https://api.artsy.net/api/artists/milton-resnick
error milton-resnick
alice-baber
https://api.artsy.net/api/artists/alice-baber
gwenn-thomas
https://api.artsy.net/api/artists/gwenn-thomas
warren-brandt
https://api.artsy.net/api/artists/warren-brandt
dennis-ashbaugh
https://api.artsy.net/api/artists/dennis-ashbaugh
roy-lichtenstein
https://api.artsy.net/api/artists/roy-lichtenstein
joan-witek
https://api.artsy.net/api/artists/joan-witek
frank-bowling
https://api.artsy.net/api/artists/frank-bowling

error jacques-hartmann
red-grooms
https://api.artsy.net/api/artists/red-grooms
dennis-smith
https://api.artsy.net/api/artists/dennis-smith
joan-miró
https://api.artsy.net/api/artists/joan-miró
error joan-miró
leonard-koscianski
https://api.artsy.net/api/artists/leonard-koscianski
james-g-davis
https://api.artsy.net/api/artists/james-g-davis
walter-hatke
https://api.artsy.net/api/artists/walter-hatke
error walter-hatke
dona-nelson
https://api.artsy.net/api/artists/dona-nelson
alan-shields
https://api.artsy.net/api/artists/alan-shields
mark-wethli
https://api.artsy.net/api/artists/mark-wethli
error mark-wethli
ralph-rosenborg
https://api.artsy.net/api/artists/ralph-rosenborg
error ralph-rosenborg
paul-klee
https://api.artsy.net/api/artists/paul-klee
emil-bisttram
https://api.artsy.net/api/artists/emil-bisttram
ed-garman
https://api.artsy.net/api/artists/ed-garman
error ed-garman
robert-helm
https://api.artsy.net/api/artists/robert-helm
nicholas-africano
https://api.artsy.net/api/artists/

paul-manes
https://api.artsy.net/api/artists/paul-manes
error paul-manes
john-bellany
https://api.artsy.net/api/artists/john-bellany
eda-mirsky-mann
https://api.artsy.net/api/artists/eda-mirsky-mann
error eda-mirsky-mann
katherine-bowling
https://api.artsy.net/api/artists/katherine-bowling
neil-jenney
https://api.artsy.net/api/artists/neil-jenney
domenic-cretara
https://api.artsy.net/api/artists/domenic-cretara
error domenic-cretara
jim-sullivan
https://api.artsy.net/api/artists/jim-sullivan
peter-waite
https://api.artsy.net/api/artists/peter-waite
robert-mason
https://api.artsy.net/api/artists/robert-mason
error robert-mason
sabina-ott
https://api.artsy.net/api/artists/sabina-ott
angelo-ippolito
https://api.artsy.net/api/artists/angelo-ippolito
error angelo-ippolito
lydia-dona
https://api.artsy.net/api/artists/lydia-dona
yuri-kuper
https://api.artsy.net/api/artists/yuri-kuper
donald-shambroom
https://api.artsy.net/api/artists/donald-shambroom
error donald-shambroom
david-krueger
https

error boyd-and-evans|fionnuala-boyd|leslie-evans
james-d-butler
https://api.artsy.net/api/artists/james-d-butler
error james-d-butler
alexander-brook
https://api.artsy.net/api/artists/alexander-brook
arthur-dove
https://api.artsy.net/api/artists/arthur-dove
charles-sheeler
https://api.artsy.net/api/artists/charles-sheeler
error charles-sheeler
beauford-delaney
https://api.artsy.net/api/artists/beauford-delaney
error beauford-delaney
stephen-hannock
https://api.artsy.net/api/artists/stephen-hannock
fred-williams
https://api.artsy.net/api/artists/fred-williams
jennifer-bartlett
https://api.artsy.net/api/artists/jennifer-bartlett
error jennifer-bartlett
susan-rothenberg
https://api.artsy.net/api/artists/susan-rothenberg
rosalie-gascoigne
https://api.artsy.net/api/artists/rosalie-gascoigne
euan-macleod
https://api.artsy.net/api/artists/euan-macleod
error euan-macleod
imants-tillers
https://api.artsy.net/api/artists/imants-tillers
joseph-delaney
https://api.artsy.net/api/artists/joseph-dela

error ernest-blumenschein
maurice-sterne
https://api.artsy.net/api/artists/maurice-sterne
clarence-holbrook-carter
https://api.artsy.net/api/artists/clarence-holbrook-carter
waldo-peirce
https://api.artsy.net/api/artists/waldo-peirce
error waldo-peirce
doris-emrick-lee
https://api.artsy.net/api/artists/doris-emrick-lee
paul-sample
https://api.artsy.net/api/artists/paul-sample
george-biddle
https://api.artsy.net/api/artists/george-biddle
error george-biddle
edwin-dickinson
https://api.artsy.net/api/artists/edwin-dickinson
richard-blow
https://api.artsy.net/api/artists/richard-blow
hobson-pittman
https://api.artsy.net/api/artists/hobson-pittman
frank-mechau
https://api.artsy.net/api/artists/frank-mechau
dale-nichols
https://api.artsy.net/api/artists/dale-nichols
error dale-nichols
mary-hoover-aiken
https://api.artsy.net/api/artists/mary-hoover-aiken
error mary-hoover-aiken
maud-morgan
https://api.artsy.net/api/artists/maud-morgan
audrey-buller
https://api.artsy.net/api/artists/audrey-bul

john-hultberg
https://api.artsy.net/api/artists/john-hultberg
john-heliker
https://api.artsy.net/api/artists/john-heliker
error john-heliker
nicolas-carone
https://api.artsy.net/api/artists/nicolas-carone
error nicolas-carone
larry-rivers
https://api.artsy.net/api/artists/larry-rivers
stephen-etnier
https://api.artsy.net/api/artists/stephen-etnier
arshile-gorky
https://api.artsy.net/api/artists/arshile-gorky
error arshile-gorky
lodewijk-karel-bruckman
https://api.artsy.net/api/artists/lodewijk-karel-bruckman
error lodewijk-karel-bruckman
carroll-cloar
https://api.artsy.net/api/artists/carroll-cloar
gerald-leslie-brockhurst
https://api.artsy.net/api/artists/gerald-leslie-brockhurst
george-tooker
https://api.artsy.net/api/artists/george-tooker
error george-tooker
nancy-ellen-craig
https://api.artsy.net/api/artists/nancy-ellen-craig
carl-a-morris
https://api.artsy.net/api/artists/carl-a-morris
error carl-a-morris
aaron-a-shikler
https://api.artsy.net/api/artists/aaron-a-shikler
error aaro

pat-steir
https://api.artsy.net/api/artists/pat-steir
joseph-mellor-hanson
https://api.artsy.net/api/artists/joseph-mellor-hanson
error joseph-mellor-hanson
cecily-brown
https://api.artsy.net/api/artists/cecily-brown
error cecily-brown
peter-sacks
https://api.artsy.net/api/artists/peter-sacks
algernon-newton-ra
https://api.artsy.net/api/artists/algernon-newton-ra
error algernon-newton-ra
george-condo
https://api.artsy.net/api/artists/george-condo
sam-francis
https://api.artsy.net/api/artists/sam-francis
error sam-francis
dennis-hopper
https://api.artsy.net/api/artists/dennis-hopper
jenny-saville
https://api.artsy.net/api/artists/jenny-saville
amy-bennett
https://api.artsy.net/api/artists/amy-bennett
error amy-bennett
richard-upton
https://api.artsy.net/api/artists/richard-upton
error richard-upton
al-loving
https://api.artsy.net/api/artists/al-loving
sylvia-plimack-mangold
https://api.artsy.net/api/artists/sylvia-plimack-mangold
rashid-rana
https://api.artsy.net/api/artists/rashid-rana

carmen-herrera
https://api.artsy.net/api/artists/carmen-herrera
amy-sillman
https://api.artsy.net/api/artists/amy-sillman
regina-bogat
https://api.artsy.net/api/artists/regina-bogat
lee-mullican
https://api.artsy.net/api/artists/lee-mullican
francis-alÿs
https://api.artsy.net/api/artists/francis-alÿs
error francis-alÿs
jacqueline-humphries
https://api.artsy.net/api/artists/jacqueline-humphries
error jacqueline-humphries
kg-subramanyan
https://api.artsy.net/api/artists/kg-subramanyan
tomie-ohtake
https://api.artsy.net/api/artists/tomie-ohtake
prabhavathi-meppayil
https://api.artsy.net/api/artists/prabhavathi-meppayil
error prabhavathi-meppayil
jesse-howard
https://api.artsy.net/api/artists/jesse-howard
error jesse-howard
adrian-ghenie
https://api.artsy.net/api/artists/adrian-ghenie
william-bailey
https://api.artsy.net/api/artists/william-bailey
stanley-whitney
https://api.artsy.net/api/artists/stanley-whitney
error stanley-whitney
steve-wheeler
https://api.artsy.net/api/artists/steve-wh

In [18]:
new = pd.DataFrame.from_dict(case_list)
new

artist  gender nationality birthday  \
0          salvador-dali    male     Spanish     1904   
1    william-rothenstein    male     British     1872   
2         george-bellows    male    American     1882   
3         paul-dougherty    male                 1877   
4              jonas-lie    male    American     1880   
..                   ...     ...         ...      ...   
578           jonas-wood    male    American     1977   
579       vivan-sundaram    male      Indian     1943   
580      harvey-quaytman    male    American     1937   
581           raqib-shaw    male      Indian     1974   
582         julia-rommel  female    American     1980   

                        hometown  
0                Figueres, Spain  
1        Bradford, Yorkshire, UK  
2    Columbus, OH, United States  
3                                 
4                                 
..                           ...  
578    Boston, MA, United States  
579                Shimla, India  
580           New York, New York  
581               Kolkata, India  
582     Salisbury, United States  

[583 rows x 5 columns]

In [19]:
bish = pd.merge(df,new,on='artist',how='inner')

### We were able to get the gender of 1203 out of the 1872 artworks we were looking at. 

In [20]:
bish

Title  Artist_Display_Name  \
0                           Berthe David-Weill        Salvador Dalí   
1                                      Madonna        Salvador Dalí   
2              Crucifixion (Corpus Hypercubus)        Salvador Dalí   
3                 The Accommodations of Desire        Salvador Dalí   
4                                Self-Portrait  William Rothenstein   
...                                        ...                  ...   
1198                     Matisse Landscape Pot           Jonas Wood   
1199  From Persian Miniatures to Stan Brakhage       Vivan Sundaram   
1200                                  Untitled      Harvey Quaytman   
1201           Ode to the Valley of Wonderment           Raqib Shaw   
1202                               Cool Topics         Julia Rommel   

     Artist_Gender Object_Name                   Department  \
0              NaN    Painting  Modern and Contemporary Art   
1              NaN    Painting  Modern and Contemporary Art   
2              NaN    Painting  Modern and Contemporary Art   
3              NaN    Painting  Modern and Contemporary Art   
4              NaN    Painting  Modern and Contemporary Art   
...            ...         ...                          ...   
1198           NaN    Painting  Modern and Contemporary Art   
1199           NaN    Painting  Modern and Contemporary Art   
1200           NaN    Painting  Modern and Contemporary Art   
1201           NaN    Painting  Modern and Contemporary Art   
1202           NaN    Painting  Modern and Contemporary Art   

                   artist  gender nationality birthday  \
0           salvador-dali    male     Spanish     1904   
1           salvador-dali    male     Spanish     1904   
2           salvador-dali    male     Spanish     1904   
3           salvador-dali    male     Spanish     1904   
4     william-rothenstein    male     British     1872   
...                   ...     ...         ...      ...   
1198           jonas-wood    male    American     1977   
1199       vivan-sundaram    male      Indian     1943   
1200      harvey-quaytman    male    American     1937   
1201           raqib-shaw    male      Indian     1974   
1202         julia-rommel  female    American     1980   

                       hometown  
0               Figueres, Spain  
1               Figueres, Spain  
2               Figueres, Spain  
3               Figueres, Spain  
4       Bradford, Yorkshire, UK  
...                         ...  
1198  Boston, MA, United States  
1199              Shimla, India  
1200         New York, New York  
1201             Kolkata, India  
1202   Salisbury, United States  

[1203 rows x 10 columns]

### How many are female?

In [21]:
bish.gender.value_counts()

male      1002
female     146
             4
Name: gender, dtype: int64